endpoint.go

```diff
@@ -34,11 +35,37 @@ type CreateEndpointRequest struct {
 	RateLimitDuration string `json:"rate_limit_duration,omitempty"`
 }
 
+// customString is a reusable type to handle JSON values that can be either strings or integers.
+// It unmarshals JSON data into a string representation, regardless of whether the input is a string or an integer.
+type customString string
+
+func (c *customString) UnmarshalJSON(b []byte) error {
+	var strValue string
+	var intValue int
+
+	if err := json.Unmarshal(b, &strValue); err == nil {
+		*c = customString(strValue)
+		return nil
+	}
+
+	if err := json.Unmarshal(b, &intValue); err == nil {
+		*c = customString(fmt.Sprintf("%d", intValue))
+		return nil
+	}
+
+	return fmt.Errorf("customstring: cannot unmarshal %v into Go value", string(b))
+}
+
+func (c *customString) MarshalJSON() ([]byte, error) {
+	return json.Marshal(string(*c))
```

Comment:
Actually on second thought, does it make sense to return these as integers instead as specified in the API?

![image](https://github.com/frain-dev/convoy-go/assets/80865609/fc113452-0b2a-42d1-bfba-b842841f4131)